src/test/index.test.ts

```diff
@@ -99,5 +104,96 @@ describe('simple-env', () => {
       expect(Object.getOwnPropertyDescriptors(env)).not.toHaveProperty('something');
       expect(Object.getOwnPropertyDescriptors(env)).toHaveProperty('somethingElse');
     });
+
+    it('will invoke the parser is loadDotEnv is true', () => {
+      existsSyncSpy.mockReturnValue(true);
+      readFileSpy.mockImplementation(() => Buffer.from('TEST=test'));
+
+      setEnv({ optional: { something: 'SOMETHING' }, options: { loadDotEnv: true } });
+
+      expect(process.env).toHaveProperty('TEST');
+    });
+
+    it('will not invoke the parser by default', () => {
+      existsSyncSpy.mockReturnValue(true);
+      readFileSpy.mockImplementation(() => Buffer.from('TEST=test'));
+
+      setEnv({ optional: { something: 'SOMETHING' } });
+
+      expect(process.env).not.toHaveProperty('TEST');
+    });
+  });
+
+  describe('parser', () => {
```

Comment:
Might make sense to move this into its own file for consistency, let me know what you think!

Reply:
Agreed, the `test` folder should mimic the `src` folder